### PR TITLE
Add List() to pls client interface

### PR DESCRIPTION
**Pull request by jwtty**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Add List() to Azure privatelinkservice client interface
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azureclients/privatelinkserviceclient/azure_privatelinkserviceclient.go [hunk 1]
```diff
@@ -215,6 +217,78 @@ func (c *Client) getPLS(ctx context.Context, resourceGroupName string, privateLi
...
+
+	return result, nil
+}
+
+// listPLS gets a list of PrivateLinkServices in the resource group.
+func (c *Client) listPLS(ctx context.Context, resourceGroupName string) ([]network.PrivateLinkService, *retry.Error) {
+	resourceID := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/%s",
+		autorest.Encode("path", c.subscriptionID),
+		autorest.Encode("path", resourceGroupName),
+		PLSResourceKinkd)
```
Comment (feiskyer) [inline pkg/azureclients/privatelinkserviceclient/azure_privatelinkserviceclient.go:256]: typo `PLSResourceKinkd`?

Comment (jwtty) [reply pkg/azureclients/privatelinkserviceclient/azure_privatelinkserviceclient.go]: Thanks for pointing out. Fixed.

**Pull request merged**